authorization-common/src/main/java/com/sequenceiq/authorization/service/model/And.java

```diff
@@ -0,0 +1,95 @@
+package com.sequenceiq.authorization.service.model;
+
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toList;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import com.sequenceiq.authorization.resource.AuthorizationResourceAction;
+
+public class And implements Authorization {
```

Comment:
Maybe you should consider to rename these classes. They make sense for devs who is in the development of this feature, but other devs won't understand it at first. These names `And`, `Or`, etc. are very general. The same applies for `Authorization` interface.

Reply:
Okay I will rename those to `AuthorizationRule`, `AllMatch` and `AnyMatch`, although I would not give them thirty letters long names just to make it clear (actually more frustrating to read) every time, in every cases that this is about authorization.